extension/src/background/helpers/dataStorage.ts

```diff
@@ -0,0 +1,36 @@
+import { browser } from "webextension-polyfill-ts";
+
+interface SetItemParams {
+  [key: string]: any;
+}
+
+export const dataStorage = {
+  getItem: async (key: null | string | string[] | { [s: string]: any }) => {
+    // Passing an object (for ex: {myKey: "aValue"}) will return the key's (myKey) value from storage.
+    // If the key is not found in storage, it will return the the value passed along in object as a default (in this case, "aValue")
+
+    const storageResult = await browser.storage.local.get(key);
+
+    return storageResult;
+  },
+  setItem: async (setItemParams: SetItemParams) => {
+    await browser.storage.local.set(setItemParams);
+  },
+};
+
+export const migrateLocalStorageToBrowserStorage = async () => {
+  const storage: { [key: string]: any } = {};
+  Object.entries(localStorage).forEach(([k, v]) => {
+    let value = v;
+    try {
+      const parsedValue = JSON.parse(v);
+      value = parsedValue;
+    } catch (e) {
+      // do not transform v
+    }
+
+    storage[k] = value;
+  });
+
+  await dataStorage.setItem(storage);
```

Comment:
this function migrates everything from localStorage to dataStorage. We'll run this every so often to keep the 2 data stores in sync (for the time being)